lib/chef/azure/service.rb

```diff
@@ -6,74 +6,45 @@ class ChefService
   include ChefAzure::Shared
   AZURE_CHEF_SERVICE_PID_FILE = "azure-chef-client.pid"
   AZURE_CHEF_CRON_NAME = 'azure_chef_extension'
+  DEFAULT_CHEF_SERVICE_INTERVAL = 30
+  CLIENT_RB_INTERVAL_ATTRIBUTE_NAME = 'interval'
 
-  # TODO - make these methods idempotent
-  def install(log_location)
-    log_location = log_location || bootstrap_directory # example default logs go to C:\chef\
-    exit_code = 0
-    message = "success"
-    error_message = "Error installing chef-client service"
-    begin
-      if windows?
-        puts "#{Time.now} Getting chef-client service status"
-        status = shell_out("sc.exe query chef-client")
-        if status.exitstatus == 1060 && status.stdout.include?("The specified service does not exist as an installed service.")
-          puts "#{Time.now} Installing chef-client service..."
-          params = " -a install -c #{bootstrap_directory}\\client.rb -L #{log_location}\\chef-client.log "
-          result = shell_out("chef-service-manager #{params}")
-          result.error!
-          puts "#{Time.now} Installed chef-client service."
-        else
-          status.error!
-          puts "#{Time.now} chef-client service is already installed."
-        end
-      end
-      # Unix - only start chef-client in daemonize mode using self.enable
-    rescue => e
-      Chef::Log.error "#{error_message} (#{e})"
-      message = "#{error_message}- #{e} - Check log file for details"
-      exit_code = 1
-    end
-    [exit_code, message]
-  end
-
-  def enable(extension_root, bootstrap_directory, log_location)
+  def enable(extension_root, log_location, chef_service_interval = DEFAULT_CHEF_SERVICE_INTERVAL)

```

Comment:
Overall it feels like there is too much logic in this method and it could be broken up into:
`install`
`start`
`stop`
`set_interval`
Then let enable.rb determine which methods to call.